Civi/Core/SqlTriggers/TimestampTriggers.php

```diff
@@ -0,0 +1,334 @@
+   *   An entity name (from civicrm_custom_group.extends).
+   *   Ex: 'Contact', 'Activity'.
+   */
+  private $customDataEntity;
+
+  /**
+   * @var string
+   *   SQL column name.
+   *   Ex: 'created_date'.
+   */
+  private $createdDate;
+
+  /**
+   * @var string
+   *   SQL column name.
+   *   Ex: 'modified_date'.
+   */
+  private $modifiedDate;
+
+  /**
+   * @var array
+   *   Ex: $relations[0] == array('table' => 'civicrm_bar', 'column' => 'foo_id');
+   */
+  private $relations;
+
+  /**
+   * @param string $tableName
+   *   SQL table name.
+   *   Ex: 'civicrm_contact', 'civicrm_activity'.
+   * @param string $customDataEntity
+   *   An entity name (from civicrm_custom_group.extends).
+   *   Ex: 'Contact', 'Activity'.
+   * @return TimestampTriggers
+   */
+  public static function create($tableName, $customDataEntity) {
+    return new static($tableName, $customDataEntity);
+  }
+
+  /**
+   * TimestampTriggers constructor.
+   *
+   * @param string $tableName
+   *   SQL table name.
+   *   Ex: 'civicrm_contact', 'civicrm_activity'.
+   * @param string $customDataEntity
+   *   An entity name (from civicrm_custom_group.extends).
+   *   Ex: 'Contact', 'Activity'.
+   * @param string $createdDate
+   *   SQL column name.
+   *   Ex: 'created_date'.
+   * @param string $modifiedDate
+   *   SQL column name.
+   *   Ex: 'modified_date'.
+   * @param array $relations
+   *   Ex: $relations[0] == array('table' => 'civicrm_bar', 'column' => 'foo_id');
+   */
+  public function __construct(
+    $tableName,
+    $customDataEntity,
+    $createdDate = 'created_date',
+    $modifiedDate = 'modified_date',
+    $relations = array()
+  ) {
+    $this->tableName = $tableName;
+    $this->customDataEntity = $customDataEntity;
+    $this->createdDate = $createdDate;
+    $this->modifiedDate = $modifiedDate;
+    $this->relations = $relations;
+  }
+
+  /**
+   * @param \Civi\Core\Event\GenericHookEvent $e
+   * @see \CRM_Utils_Hook::triggerInfo
+   */
+  public function onTriggerInfo($e) {
+    $this->alterTriggerInfo($e->info, $e->tableName);
+  }
+
+  /**
+   * Get a list of triggers for the contact table.
+   *
+   * @see \CRM_Utils_Hook::triggerInfo
+   * @see \CRM_Core_DAO::triggerRebuild
+   * @see http://issues.civicrm.org/jira/browse/CRM-10554
+   *
+   * @param $info
+   * @param null $tableFilter
+   */
+  public function alterTriggerInfo(&$info, $tableFilter = NULL) {
+    //during upgrade, first check for valid version and then create triggers
+    //i.e the columns created_date and modified_date are introduced in 4.3.alpha1 so dont create triggers for older version
+    if (\CRM_Core_Config::isUpgradeMode()) {
+      if (!\CRM_Core_DAO::checkFieldExists($this->getTableName(),
+        $this->getCreatedDate())
+      ) {
+        return;
+      }
+      //      $currentVer = CRM_Core_BAO_Domain::version(TRUE);
+      //      if (version_compare($currentVer, '4.3.alpha1') < 0) {
+      //        return;
+      //      }
```

Comment:
Analytically, you're right, the commented code is superfluous and may confuse future readers. Emotionally, based on some of the history, I wanted to keep the example handy for a month. But it's not that important. Reworking the comments.